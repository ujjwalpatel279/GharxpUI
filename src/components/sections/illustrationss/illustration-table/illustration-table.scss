.c-illustration-table-download-button {
  padding-right: 0.25rem;
}

.c-illustration-table-edit-button {
  padding-right: 0.25rem;
}
