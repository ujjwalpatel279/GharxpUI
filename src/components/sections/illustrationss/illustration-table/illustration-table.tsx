import React, { FunctionComponent, useEffect, useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import { EsisSummaryInterface } from 'shared/models';
import { Grid, Icon } from '../../../shared';
import { NoGridData } from '../../../shared';
import { GridColumn, GridRow } from '../../../shared/grid/grid';
import './illustration-table.scss';

interface illustrationTableProps {
  esisSummary: EsisSummaryInterface[];
  editPage: string;
}

export const IllustrationTable: FunctionComponent<illustrationTableProps> = ({ esisSummary, editPage }) => {
  const [illustrationDocuments, setIllustrationDocuments] = useState<[]>([]);
  const columns: GridColumn[] = [
    {
      id: 'description',
      description: 'Product',
    },
    {
      id: 'mortgageTerm',
      description: 'Product Term',
    },
    {
      id: 'productTypeId',
      description: 'Product Type',
      sort: false,
      render(column: GridColumn, row: GridRow) {
        return <>{row[column.id] === 1 ? 'Fixed' : 'Variable'}</>;
      },
    },
    {
      id: 'standardVariableRate',
      description: 'Subsequent Rate',
      render(column: GridColumn, row: GridRow) {
        return <>{`${row[column.id]}%`}</>;
      },
    },
    {
      id: 'minimumLoanToValue',
      description: 'LTV',
      render(column: GridColumn, row: GridRow) {
        return <>{`${row.minimumLoanToValue} - ${row.maximumLoanToValue}%`}</>;
      },
    },
    {
      id: 'download',
      description: '',
      sort: false,
      render(column: GridColumn, row: GridRow) {
        return (
          <Link to="/" className="c-btn c-btn--secondary">
            <div className="c-illustration-table-download-button">
              <Icon name="download"></Icon>
            </div>
            Download
          </Link>
        );
      },
    },
    {
      id: 'edit',
      description: '',
      sort: false,
      render(column: GridColumn, row: GridRow) {
        return (
          <Link to={editPage ?? '/'} className="c-btn c-btn--secondary">
            <div className="c-illustration-table-edit-button">
              <Icon name="illustration" colour="primary"></Icon>
            </div>
            Edit
          </Link>
        );
      },
    },
  ];
  return (
    <React.Fragment>
      {esisSummary && (
        <Grid
          columns={columns}
          rowData={esisSummary}
          noRowsMessage={<NoGridData image="no-clients" message="You currently have no illustrations" />}
          showHeaders={esisSummary.length > 0}
          defaultDirection="Desc"
          defaultSortColumn="applicationId"
        />
      )}
    </React.Fragment>
  );
};
