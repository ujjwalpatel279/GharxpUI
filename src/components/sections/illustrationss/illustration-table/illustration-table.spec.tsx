import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { IllustrationTable } from './illustration-table';

describe('Illustration Component', () => {
  afterEach(cleanup);

  it('renders the illustration component', () => {
    const { container } = render(<IllustrationTable esisSummary={[]} editPage={''}></IllustrationTable>);
    expect(container).toMatchSnapshot();
  });
});
