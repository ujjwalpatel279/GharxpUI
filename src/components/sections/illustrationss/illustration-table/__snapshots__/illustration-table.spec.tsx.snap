// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Illustration Component renders the illustration component 1`] = `
<div>
  <div
    class="c-overflow"
  >
    <table
      class="c-grid"
    >
      <thead>
        <tr>
          <th
            colspan="1"
            hidden=""
          >
            <button>
              Product
            </button>
          </th>
          <th
            colspan="1"
            hidden=""
          >
            <button>
              Product Term
            </button>
          </th>
          <th
            colspan="1"
            hidden=""
          >
            Product Type
          </th>
          <th
            colspan="1"
            hidden=""
          >
            <button>
              Subsequent Rate
            </button>
          </th>
          <th
            colspan="1"
            hidden=""
          >
            <button>
              LTV
            </button>
          </th>
          <th
            colspan="1"
            hidden=""
          />
          <th
            colspan="1"
            hidden=""
          />
        </tr>
      </thead>
      <tbody>
        <tr
          class="c-grid__no-data"
        >
          <td
            colspan="7"
          >
            <div
              class="c-grid-no-data"
            >
              <img
                alt="no-data"
                class="c-grid-no-data__image"
                src="[object Object]"
              />
              <div
                class="c-grid-no-data__message"
              >
                You currently have no illustrations
                 
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
